Erros ao executar `npm run build` contra o codigo gerado.

# Erro:

```bash
npm run build

> Person@1.0.0 prebuild
> rimraf dist


> Person@1.0.0 build
> nest build

src/app/entities/born_corporation.ts:55:18 - error TS2339: Property 'id' does not exist on type 'BornCorporationEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/born_enterprise.ts:55:18 - error TS2339: Property 'id' does not exist on type 'BornEnterpriseEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/cellphone_corporation.ts:55:18 - error TS2339: Property 'id' does not exist on type 'CellphoneCorporationEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/cellphone_enterprise.ts:55:18 - error TS2339: Property 'id' does not exist on type 'CellphoneEnterpriseEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/cpf_corporation.ts:55:18 - error TS2339: Property 'id' does not exist on type 'CpfCorporationEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/cpf_enterprise.ts:55:18 - error TS2339: Property 'id' does not exist on type 'CpfEnterpriseEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/email_corporation.ts:55:18 - error TS2339: Property 'id' does not exist on type 'EmailCorporationEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/email_enterprise.ts:55:18 - error TS2339: Property 'id' does not exist on type 'EmailEnterpriseEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/gender_corporation.ts:55:18 - error TS2339: Property 'id' does not exist on type 'GenderCorporationEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/gender_enterprise.ts:55:18 - error TS2339: Property 'id' does not exist on type 'GenderEnterpriseEntity'.

55   return `${this.id} - ${this.updated_at} `;
                    ~~
src/app/entities/person.ts:49:31 - error TS2339: Property 'external' does not exist on type 'PersonEntity'.

49   return `${this.id} - ${this.external} `;
                                 ~~~~~~~~

Found 11 error(s).
```

--- a/src/db.reader.postgres.ts
+++ b/src/db.reader.postgres.ts
@@ -38,40 +38,52 @@ export class DbReader {
       const schemaInfo: Table[] = [];
 
       for (const tableName of tables) {
-        const columnsQuery = `
-          SELECT 
-            c.column_name, 
-            c.data_type, 
-            c.character_maximum_length, 
-            c.is_nullable, 
-            c.column_default,
-            pgd.description AS column_comment
-          FROM 
-            information_schema.columns c
-          LEFT JOIN 
-            pg_catalog.pg_statio_all_tables as st on c.table_schema = st.schemaname and c.table_name = st.relname
-          LEFT JOIN 
-            pg_catalog.pg_description pgd on pgd.objoid = st.relid and pgd.objsubid = c.ordinal_position
-          WHERE 
-            c.table_schema = 'public' AND c.table_name = $1
-        `;
+		const columnsQuery = `
+		SELECT
+		  c.column_name,
+		  c.data_type,
+		  c.character_maximum_length,
+		  c.is_nullable,
+		  c.column_default,
+		  pgd.description AS column_comment,
+		  (SELECT EXISTS (
+			SELECT 1
+			FROM information_schema.table_constraints tc
+			JOIN information_schema.key_column_usage kcu
+			ON tc.constraint_name = kcu.constraint_name
+			AND tc.table_schema = kcu.table_schema
+			WHERE tc.constraint_type = 'PRIMARY KEY'
+			AND tc.table_name = c.table_name
+			AND kcu.column_name = c.column_name
+		  )) AS is_primary_key
+		FROM
+		  information_schema.columns c
+		LEFT JOIN
+		  pg_catalog.pg_statio_all_tables as st on c.table_schema = st.schemaname and c.table_name = st.relname
+		LEFT JOIN
+		  pg_catalog.pg_description pgd on pgd.objoid = st.relid and pgd.objsubid = c.ordinal_position
+		WHERE
+		  c.table_schema = 'public' AND c.table_name = $1
+	  `;
         const columnsResult = await client.query<{
           column_name: string;
           data_type: string;
           character_maximum_length: number | null;
           is_nullable: 'YES' | 'NO';
+		  is_primary_key: boolean;
           column_default: string | null;
           column_comment: string | null;
         }>(columnsQuery, [tableName]);
 
         const columns: Column[] = columnsResult.rows.map(column => ({
-          columnName: column.column_name,
-          dataType: column.data_type,
-          characterMaximumLength: column.character_maximum_length,
-          isNullable: column.is_nullable === 'YES',
-          columnDefault: column.column_default,
-          columnComment: column.column_comment,
-        }));
+			columnName: column.column_name,
+			dataType: column.data_type,
+			characterMaximumLength: column.character_maximum_length,
+			isNullable: column.is_nullable === 'YES',
+			isPrimaryKey: column.is_primary_key,
+			columnDefault: column.column_default,
+			columnComment: column.column_comment
+		  }));
 
         const relationsQuery = `
           SELECT

--- a/src/interfaces.ts
+++ b/src/interfaces.ts
@@ -5,6 +5,7 @@ export interface Column {
   dataType: string;
   characterMaximumLength: number | null;
   isNullable: boolean;
+  isPrimaryKey: boolean;
   columnDefault: string | null;
   columnComment: string | null;
 }
@@ -42,7 +43,7 @@ export interface DbReaderConfig {
     'env' |
     'package.json' |
     'readme' |
-    'datasource' | 
+    'datasource' |
     'diagram'
   ];
 }

--- a/src/static-templates.ts
+++ b/src/static-templates.ts
@@ -1,24 +1,21 @@
 // src/static-templates.ts
 
 export const entityTemplate = (
-  tableName: string,
-  columns: string,
-  relations: string,
-  imports: string,
-  customMethods: string = ''
-) => `
-import { Entity, Column, PrimaryGeneratedColumn, ManyToOne, OneToOne, OneToMany, ManyToMany, JoinColumn, CreateDateColumn, UpdateDateColumn, DeleteDateColumn } from 'typeorm';
-import 'reflect-metadata';
-import { ApiProperty } from '@nestjs/swagger';
-${imports}
+	tableName: string,
+	columns: string,
+	relations: string,
+	imports: string,
+	customMethods: string = ''
+  ) => `
+  ${imports}
 
-@Entity('${tableName}')
-export class ${toPascalCase(tableName)}Entity {
-${columns}
-${relations}
+  @Entity('${tableName}')
+  export class ${toPascalCase(tableName)}Entity {
+	${columns}
+	${relations}
 
-${customMethods}
-}`;
+	${customMethods}
+  }`;
 
 export const columnTemplate = (
   columnDecorator: string,

--- a/src/typeorm-entity-generator.ts
+++ b/src/typeorm-entity-generator.ts
@@ -1,3 +1,5 @@
+
+// src/typeorm-entity-generator.ts
 import * as fs from 'fs';
 import * as path from 'path';
 import { Table, Column, Relation, DbReaderConfig } from './interfaces';
@@ -31,45 +33,49 @@ export class TypeORMEntityGenerator {
   }
 
   private generateEntityContent(table: Table): string {
-    const columns = table.columns
-      .filter(col => !this.isRelationColumn(col.columnName, table.relations))
-      .map(col => this.generateColumnDefinition(col))
-      .join('\n  ');
+	const primaryKeys = table.columns.filter(col => col.columnName.includes('id')); // Identifica colunas de chave primária (exemplo simplificado)
 
-    const relations = table.relations.map(rel => this.generateRelationDefinition(rel)).join('\n  ');
-    const imports = this.generateImports(table);
-    const customMethods = this.generateCustomMethods(table);
+	const columns = table.columns
+	  .filter(col => !this.isRelationColumn(col.columnName, table.relations))
+	  .map(col => this.generateColumnDefinition(col, primaryKeys.includes(col)))
+	  .join('\n  ');
 
-    return entityTemplate(table.tableName, columns, relations, imports, customMethods);
+	const relations = table.relations.map(rel => this.generateRelationDefinition(rel)).join('\n  ');
+	const imports = this.generateImports(table);
+	const customMethods = this.generateCustomMethods(table);
+
+	return entityTemplate(table.tableName, columns, relations, imports, customMethods);
   }
 
-  private generateColumnDefinition(column: Column): string {
-	const options: string[] = [];
-	const typeOptions: string[] = [];
+  private generateColumnDefinition(column: Column, isPrimaryKey: boolean = false): string {
+    const options: string[] = [];
+    const typeOptions: string[] = [];
 
-	if (column.isNullable) typeOptions.push('nullable: true');
-	if (column.columnDefault) options.push(`default: "${column.columnDefault.replace(/"/g, '\\"')}"`);
-	if (column.characterMaximumLength) options.push(`length: ${column.characterMaximumLength}`);
+    if (column.isNullable) typeOptions.push('nullable: true');
+    if (column.columnDefault) options.push(`default: "${column.columnDefault.replace(/"/g, '\\"')}"`);
+    if (column.characterMaximumLength) options.push(`length: ${column.characterMaximumLength}`);
 
-	let columnDecorator = column.isPrimaryKey
-	  ? `@PrimaryColumn({ type: '${typeMapping[column.dataType] || column.dataType}' })`
-	  : `@Column({ type: '${typeMapping[column.dataType] || column.dataType}', ${options.join(', ')} })`;
+    let columnDecorator = `@Column({ type: '${typeMapping[column.dataType] || column.dataType}', ${options.join(', ')} })`;
 
-	if (column.columnName === 'created_at') {
-	  columnDecorator = `@CreateDateColumn()`;
-	}
+    if (isPrimaryKey) {
+        columnDecorator = `@PrimaryColumn({ type: '${typeMapping[column.dataType] || column.dataType}', ${options.join(', ')} })`;
+    }
 
-	if (column.columnName === 'updated_at') {
-	  columnDecorator = `@UpdateDateColumn()`;
-	}
+    if (column.columnName === 'created_at') {
+        columnDecorator = `@CreateDateColumn()`;
+    }
 
-	if (column.columnName === 'deleted_at') {
-	  columnDecorator = `@DeleteDateColumn()`;
-	}
+    if (column.columnName === 'updated_at') {
+        columnDecorator = `@UpdateDateColumn()`;
+    }
 
-	const apiPropertyDecorator = `@ApiProperty({ description: "${column.columnComment || ''}", ${typeOptions.join(', ')} })`;
+    if (column.columnName === 'deleted_at') {
+        columnDecorator = `@DeleteDateColumn()`;
+    }
 
-	return columnTemplate(columnDecorator, apiPropertyDecorator, column.columnName, jsTypeMapping[column.dataType] || 'any');
+    const apiPropertyDecorator = `@ApiProperty({ description: "${column.columnComment || ''}", ${typeOptions.join(', ')} })`;
+
+    return columnTemplate(columnDecorator, apiPropertyDecorator, column.columnName, jsTypeMapping[column.dataType] || 'any');
   }
 
   private generateRelationDefinition(relation: Relation): string {
@@ -88,15 +94,29 @@ export class TypeORMEntityGenerator {
   }
 
   private generateImports(table: Table): string {
-    const imports = new Set<string>();
+	const typeormImports = new Set<string>([
+	  'Entity',
+	  'Column',
+	  'CreateDateColumn',
+	  'UpdateDateColumn',
+	  'DeleteDateColumn',
+	  'PrimaryColumn',
+	  'JoinColumn'
+	]);
 
-    table.relations.forEach(relation => {
-      imports.add(`import { ${toPascalCase(relation.foreignTableName)}Entity } from './${removeTbPrefix(relation.foreignTableName)}';`);
-    });
+	table.relations.forEach(relation => {
+	  typeormImports.add(relation.relationType);
+	});
 
-    return Array.from(imports).join('\n');
+	const entityImports = table.relations.map(relation =>
+	  `import { ${toPascalCase(relation.foreignTableName)}Entity } from './${removeTbPrefix(relation.foreignTableName)}';`
+	).join('\n');
+
+	const typeormImportsString = `import { ${Array.from(typeormImports).join(', ')} } from 'typeorm';`;
+
+	return `${typeormImportsString}\nimport 'reflect-metadata';\nimport { ApiProperty } from '@nestjs/swagger';\n${entityImports}`;
   }
-
+  
   private generateCustomMethods(table: Table): string {
 	const column = table.columns.find(col =>
         !["id", "external_id", "updated_at", "created_at", "deleted_at"].includes(col.columnName)

--- a/src/typeorm-entity-generator.ts
+++ b/src/typeorm-entity-generator.ts
@@ -44,18 +44,16 @@ export class TypeORMEntityGenerator {
   }
 
   private generateColumnDefinition(column: Column): string {
-    const options: string[] = [];
-    const typeOptions: string[] = [];
+	const options: string[] = [];
+	const typeOptions: string[] = [];
 
-    if (column.isNullable) typeOptions.push('nullable: true');
-    if (column.columnDefault) options.push(`default: "${column.columnDefault.replace(/"/g, '\\"')}"`);
-    if (column.characterMaximumLength) options.push(`length: ${column.characterMaximumLength}`);
+	if (column.isNullable) typeOptions.push('nullable: true');
+	if (column.columnDefault) options.push(`default: "${column.columnDefault.replace(/"/g, '\\"')}"`);
+	if (column.characterMaximumLength) options.push(`length: ${column.characterMaximumLength}`);
 
-    let columnDecorator = `@Column({ type: '${typeMapping[column.dataType] || column.dataType}', ${options.join(', ')} })`;
-
-	if (column.columnName === 'id') {
-	  columnDecorator = `@PrimaryGeneratedColumn()`;
-	}
+	let columnDecorator = column.isPrimaryKey
+	  ? `@PrimaryColumn({ type: '${typeMapping[column.dataType] || column.dataType}' })`
+	  : `@Column({ type: '${typeMapping[column.dataType] || column.dataType}', ${options.join(', ')} })`;
 
 	if (column.columnName === 'created_at') {
 	  columnDecorator = `@CreateDateColumn()`;
@@ -69,9 +67,9 @@ export class TypeORMEntityGenerator {
 	  columnDecorator = `@DeleteDateColumn()`;
 	}
 
-    const apiPropertyDecorator = `@ApiProperty({ description: "${column.columnComment || ''}", ${typeOptions.join(', ')} })`;
+	const apiPropertyDecorator = `@ApiProperty({ description: "${column.columnComment || ''}", ${typeOptions.join(', ')} })`;
 
-    return columnTemplate(columnDecorator, apiPropertyDecorator, column.columnName, jsTypeMapping[column.dataType] || 'any');
+	return columnTemplate(columnDecorator, apiPropertyDecorator, column.columnName, jsTypeMapping[column.dataType] || 'any');
   }
 
   private generateRelationDefinition(relation: Relation): string {

--- a/src/typeorm-entity-generator.ts
+++ b/src/typeorm-entity-generator.ts
@@ -100,9 +100,15 @@ export class TypeORMEntityGenerator {
   }
 
   private generateCustomMethods(table: Table): string {
+	const column = table.columns.find(col =>
+        !["id", "external_id", "updated_at", "created_at", "deleted_at"].includes(col.columnName)
+    );
+
+    const columnName = column ? ` - \${this.${this.removeIdSuffix(column.columnName)}}` : '';
+
     return `
     toString() {
-      return \`\${this.id} - \${this.${this.removeIdSuffix(table.columns[1].columnName)}} \`;
+      return \`\${this.external_id}${columnName}\`;
     }`;
   }
 
@@ -111,6 +117,6 @@ export class TypeORMEntityGenerator {
   }
 
   private removeIdSuffix(columnName: string): string {
-    return columnName.endsWith('_id') ? columnName.slice(0, -3) : columnName;
+    return columnName.endsWith('_id') && columnName !== 'external_id' ? columnName.slice(0, -3) : columnName;
   }
 }

